lib/chef/knife/opc_user_delete.rb

```diff
@@ -20,25 +20,36 @@
 module Opc
   class OpcUserDelete < Chef::Knife
     category "OPSCODE PRIVATE CHEF ORGANIZATION MANAGEMENT"
-    banner "knife opc user delete USERNAME [-d]"
+    banner "knife opc user delete USERNAME"
 
-    option :no_disassociate_user,
-    :long => "--no-disassociate-user",
-    :short => "-d",
-    :description => "Don't disassociate the user first"
+    deps do
+      require 'chef/json_compat'
+    end
 
     include Chef::Mixin::RootRestv0
 
     def run
       username = @name_args[0]
       ui.confirm "Do you want to delete the user #{username}"
-      unless config[:no_disassociate_user]
-         orgs =  root_rest.get("users/#{username}/organizations")
-         org_names = orgs.map {|o| o['organization']['name']}
-         org_names.each do |org|
-            ui.output root_rest.delete("organizations/#{org}/users/#{username}")
-         end
+
+      orgs = root_rest.get("users/#{username}/organizations")
+      org_names = orgs.map {|o| o['organization']['name']}
+      org_names.each do |org|
+        begin
+          ui.output root_rest.delete("organizations/#{org}/users/#{username}")
+        rescue Net::HTTPServerException => e
+          body = Chef::JSONCompat.from_json(e.response.body)
+          if e.response.code == "403" && body["error"] == "Please remove #{username} from this organization's admins group before removing him or her from the organization."
+            ui.error "Error removing user #{username} from org #{org} due to user being in that org's admins group."
+            ui.msg "Please remove #{username} from the admins group for #{org} before deleting the user."
+            ui.msg "This can be accomplished by passing --force to the org user remove command."
```

Comment:
Long term, once server-admins is complete and we move this and acl into core knife, I would like to just replace all the wrapping with `chef-server-ctl knife` and `chef-server-ctl knife-org #{org}` that just wraps Chef Server embedded knife and passes things along.